core/src/main/java/com/datastrato/gravitino/listener/PartitionEventDispatcher.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datastrato.gravitino.listener;
+
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.catalog.PartitionDispatcher;
+import com.datastrato.gravitino.exceptions.NoSuchPartitionException;
+import com.datastrato.gravitino.exceptions.PartitionAlreadyExistsException;
+import com.datastrato.gravitino.listener.api.event.AddPartitionEvent;
+import com.datastrato.gravitino.listener.api.event.AddPartitionFailureEvent;
+import com.datastrato.gravitino.listener.api.event.DropPartitionEvent;
+import com.datastrato.gravitino.listener.api.event.DropPartitionFailureEvent;
+import com.datastrato.gravitino.listener.api.event.GetPartitionEvent;
+import com.datastrato.gravitino.listener.api.event.GetPartitionFailureEvent;
+import com.datastrato.gravitino.listener.api.event.ListPartitionEvent;
+import com.datastrato.gravitino.listener.api.event.ListPartitionFailureEvent;
+import com.datastrato.gravitino.listener.api.info.PartitionInfo;
+import com.datastrato.gravitino.rel.partitions.Partition;
+import com.datastrato.gravitino.utils.PrincipalUtils;
+import com.google.common.collect.Maps;
+
+/**
+ * {@code PartitionEventDispatcher} is a decorator for {@link PartitionDispatcher} that not only
+ * delegates partition operations to the underlying partition dispatcher but also dispatches
+ * corresponding events to an {@link com.datastrato.gravitino.listener.EventBus} after each
+ * operation is completed. This allows for event-driven workflows or monitoring of partition
+ * operations.
+ */
+public class PartitionEventDispatcher implements PartitionDispatcher {
```

Comment:
missing `partitionExists` and `purgePartition` event process?

Reply:
Add ExistPartitionEvent & PurgePartitionEvent